### PR TITLE
Alias feature

### LINKED ISSUE #232: Tweek key linking
It would be nice to have the capability of creating a key which is reference for another key.
It can solve a lot of problems related to migration, allow refactoring of our key structures and can provide backward compatibility.

This feature can implemented in both Tweek level or in JPad level.
I think the first iteration should be in Tweek since it's simpler and such capability shouldn't be coupled to the rules engine.

One simple way to achieve it, is applying meta-data on the referencing key with the target key and using management service to duplicate the key definition (format and source) on bundling.

### PR DESCRIPTION
**Pull request by nataly87s**
closes #232

### CODE REVIEW

File: core/Engine/Tweek.Engine.Tests/Core/KeyAliasParserTests.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿using LanguageExt;
+using Tweek.Engine.Core.Context;
+using Xunit;
+
```
Comment (Yshayy) [inline core/Engine/Tweek.Engine.Tests/Core/KeyAliasParserTests.cs]: using static LanguageExt.Prelude

File: services/editor/src/store/ducks/selectedKey.js [hunk 2]
```diff
@@ -295,33 +294,84 @@ export function saveKey() {
...
   return async function (dispatch, getState) {
-    const { selectedKey: { key } } = getState();
+    const { selectedKey: { key, aliases } } = getState();
 
-    if (!(await dispatch(showConfirm(deleteKeyAlert(key)))).result) return;
+    if (!(await dispatch(showConfirm(deleteKeyAlert(key, aliases)))).result) return;
 
     dispatch(push('/keys'));
     try {
       await fetch(`/api/keys/${key}`, {
```
Comment (Yshayy) [inline services/editor/src/store/ducks/selectedKey.js]: Is it ok for DELETE to get body data?

File: services/git-service/BareRepository/tests-source/manifests/behavior_tests/delete_key/delete/alias.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{
+  "key_path": "behavior_tests/delete_key/delete/alias",
+  "meta": {
+      "name": "behavior_tests/delete_key/delete/alias",
```
Comment (Yshayy) [inline services/git-service/BareRepository/tests-source/manifests/behavior_tests/delete_key/delete/alias.json]: let's try to remove all meta other then archived

File: services/editor/src/store/ducks/selectedKey.js (hunk 2 above)
Comment (nataly87s) [reply services/editor/src/store/ducks/selectedKey.js]: yes
https://stackoverflow.com/questions/299628/is-an-entity-body-allowed-for-an-http-delete-request

**Pull request merged**